docs/reference-config-guide.md

```diff
@@ -105,3 +105,130 @@ spec:
     tier: frontend
     {{- end }}
 ```
+
+# Per-template configuration
+
+## Pre-merging
+
+If you don't want to check live-manifest exactly matches your template you can enable merging.
+This will do a strategic merge of the template into the manifest which will remove features not mentioned in the template from the diff.
+This can be useful when dealing with annotation or labels which may be of no consiquence to your check.
+Note that this could cover up differences in things you do care about so use it with care.
+This can be configured for a manifest by adding
+`# cluster-compare: allow-undefined-extras=true` in your template
+
+example when comparing the template:
+
+```yaml
+# cluster-compare: allow-undefined-extras=true
```

Comment:
Maybe consider a bit more "known" syntax like e.g how we have kernel args? e.g 
`# CLUSTER_COMPARE=allow_undefined_extras=true another_config=true`

With this self-contained config...so have two places to configure stuff? i.e configs inside the CR + metadata.yaml? Should be this be converged?

Reply:
I think going for a yaml syntax can be a good solution:

#cluster-compare start 
#allow-undefined-extras: true
#cluster-compare end

this will allow:
1. easy addition of fields
2. a known syntax with many options (nested values)
3. the code will be more neat because the parsing is a lot easier, and in case there are keys that are not supported in the config an error will be raised automatically ( i will add a comment with more explanations in the relevant section in the code)

Reply:
@pixelsoccupied about what you said about this:
```
With this self-contained config...so now we have two places to configure stuff? i.e configs inside the CR + metadata.yaml? Should be this be converged?
```

the role of the metadata.yaml is to set global reference settings and relations of multiple templates (grouping). this feature adds rules about only a specific template and this rule depends on the structure of the template that is why its better and more easier to maintain ,  when writing\viewing the template its easier to see all settings in the same file, and understand the structure of the template based on these values

Reply:
update:
instead of 
```
#cluster-compare start
#allow-undefined-extras: true
#cluster-compare end
```
you can do:

```
#templateConfig:
#    allow-undefined-extras: true
```

Reply:
> you can do:

we should stick to a single line config with a well-know key to simplify the custom config syntax. Otherwise to be complete we need a full blown AST. Check out of discussion here https://github.com/go-yaml/yaml/issues/132/ 

But my concern is not the syntax itself but if we can use something that's already established/well-known.


Thinking about this a bit more...can we rely on go-template for this? 

e.g 
```
{{ template "AllowUndefinedExtras"  }}
apiVersion: v1
kind: Namespace
metadata:
  name: openshift-storage
  annotations:
    workload.openshift.io/allowed: management
  labels:
    openshift.io/cluster-monitoring: "true"
```

And then we define `AllowUndefinedExtras` in go which can change modify the config?

Reply:
This seems like a good idea then we can change it later to whatever we want. Perhaps we could use a function which returns something instead of a template so it looks like 
```
{{ ClusterCompareConfigAllowUndefinedFields  }}
apiVersion: v1
kind: Namespace
metadata:
  name: openshift-storage
  annotations:
    workload.openshift.io/allowed: management
  labels:
    openshift.io/cluster-monitoring: "true"
```
that way we don't run into any templates that other people may define.

Reply:
I would also like to support something like this

```yaml
apiVersion: v1
kind: Namespace
metadata:
  name: openshift-storage
  annotations: # cluster-compare: allow-undefined-extras=true
      workload.openshift.io/allowed: management
```

Reply:
We after a discussion decided to put the config into the metadata file.